### PR TITLE
Add Power BI refresh

### PR DESCRIPTION
**Pull request by mcardosos**

### CODE REVIEW

**Review by troydai — CHANGES_REQUESTED** [review]
Let's move the logic into the service as a micro-service.

File: reportutils/powerbi.go [hunk 1]
```diff
@@ -1,105 +1,51 @@
...
-		string(username),
-		string(password),
-		"https://analysis.windows.net/powerbi/api")
-	if err != nil {
-		return a, fmt.Errorf("failed to create a new service principal token: %v", err)
-	}
-	return autorest.NewBearerAuthorizer(spt), nil
+// RefreshPowerBI requests the PowerBI service to refresh a dataset
+func RefreshPowerBI(run *models.Run) {
+	if run.IsOfficial() {
```
Comment (troydai) [inline reportutils/powerbi.go]: I would do this to reduce the indentation:
```
if !run.IsOfficial() {
    common.LogInf("....")
    return
}

// rest of the logic when the run is official
```

File: reportutils/common.go [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package reportutils
+
+import (
+	"net/http"
+)
+
+var httpClient = &http.Client{}
```
Comment (troydai) [inline reportutils/common.go]: Is it a bit overkill?

File: reportutils/powerbi.go [hunk 3]
```diff
@@ -1,105 +1,51 @@
...
+		}
+		body, err := json.Marshal(content)
+		if err != nil {
+			common.LogInfo("Fail to marshal JSON during request refreshing PowerBI.")
+			return
+		}
+
+		req, err := http.NewRequest(
+			http.MethodPost,
+			fmt.Sprintf("http://%s/powerbi", common.DNSNamePowerBIService),
```
Comment (troydai) [inline reportutils/powerbi.go]: This is not related, but I think this is a poor service resource name.

`POST /PowerBIServce/powerbi` doesn't necessarily tell the reader anything about what this action is doing. It is not a straightforward resource name. Maybe we can review the name a bit later.

**Review by troydai — CHANGES_REQUESTED** [review]

File: reportutils/powerbi.go (hunk 3 above)
Comment (mcardosos) [reply reportutils/powerbi.go]: What about `reportrefresh` ?

Comment (mcardosos) [reply reportutils/powerbi.go]: Fixed

File: reportutils/common.go (hunk 2 above)
Comment (mcardosos) [reply reportutils/common.go]: Fixed

File: Gopkg.toml [hunk 4]
```diff
@@ -21,6 +21,11 @@
 #  version = "2.4.0"
 
 
+[[constraint]]
+  name = "github.com/Azure/go-autorest"
```
Comment (troydai) [inline Gopkg.toml]: I didn't see this reference in the source code. Is it needed?

**Review by troydai — DISMISSED** [review]
Will till the service is merged and redeployed.

File: Gopkg.toml (hunk 4 above)
Comment (mcardosos) [reply Gopkg.toml]: Nope, leftover from previous implementation. Removed :)

**Comment by mcardosos** [comment]
Deployed

**Review by troydai — APPROVED** [review]

**Pull request merged**